src/mcode-pilot/utils/FilterOptions.js

```diff
@@ -0,0 +1,89 @@
+export default class FilterOptions {
+    constructor(props) {
+        this.filters = props;
+        this.filterList = this.getFiltersAsList();
+    }
+
+    getAllActiveFilters() {
+        return this.filterList.filter((filter) => {
+            return filter.selected;
+        });
+    }
+
+    getFiltersAsList() {
+        let returnArray = [];
+        Object.keys(this.filters).forEach((key) => {
+            this.recursiveFilterSearch(this.filters[key], returnArray);
+        });
+
+        return returnArray;
+    }
+
+    getAllActiveValuesByMcodeElement(asList=false) {
```

Comment:
Whats the purpose of the asList variable?  This method looks like it maps all of the active filters into a json object by the mcode type and if asList is passed in as true,  simply wraps that json object into an array.  What is the use case for this output to be used as an array as opposed to an object?